### DISCUSSION TITLE
Monorepo presets: `tokio-rs/tracing` & `tokio-rs/tokio` or orgRepo `tokio-rs`

### DISCUSSION THREAD

**Discussion by LeoniePhiline** [renovatebot/renovate#22581] [Suggest an Idea]
### What would you like Renovate to be able to do?

- Auto-combine `tokio-rs` org cargo packages in dependency updates.

#### Alternatively (see section "Discussion"):

- Auto-combine `tokio-rs/tracing` cargo packages in dependency updates.
- Auto-combine `tokio-rs/tokio` cargo packages in dependency updates.

### If you have any ideas on how this should be implemented, please tell us here.

An `orgGroup` for https://github.com/tokio-rs, akin to https://github.com/renovatebot/renovate/blob/144cf15315d1009cd1c01fd75cf9a209e3dbaedb/lib/config/presets/internal/monorepo.ts#L260-L267

Additionally, add to `group:monorepos` (https://docs.renovatebot.com/presets-group/#groupmonorepos) to automatically include in `config:base` via new group `group:tokioMonorepo`.

Resulting config:

```json
    {
      "description": "Group tokio monorepo",
      "matchManagers": ["cargo"],
      "matchSourceUrlPrefixes": ["https://github.com/tokio-rs/"],
      "groupName": "tokio monorepo"
    },
```

#### Alternative: Individual monorepos for `tokio-rs` org's monorepos:

Could be implemented by adding https://github.com/tokio-rs/tracing / https://github.com/tokio-rs/tokio to https://github.com/renovatebot/renovate/blob/144cf15315d1009cd1c01fd75cf9a209e3dbaedb/lib/config/presets/internal/monorepo.ts#L6-L256 to add tokio-rs/tracing to https://docs.renovatebot.com/presets-monorepo/.

Additionally, add to `group:monorepos` (https://docs.renovatebot.com/presets-group/#groupmonorepos) to automatically include in `config:base` via new groups `group:tokioTracingMonorepo` &  `group:tokioMonorepo`. (More precisely `group:tokioTokioMonorepo`? Using pattern `group:<org><repo>Monorepo`)

#### Discussion

While tokio crates are closely related, one might not want to have the entire org in a single renovate group.

E.g. `axum` is more unstable and will have more major versions / breaking changes than `tokio` or `tracing`.

### Is this a feature you are interested in implementing yourself?

Maybe

**Comment by HonkingGoose** [comment]
I converted this to a Discussion, as the requirements are not clear yet. Once the requirements are clear, the maintainers will copy/paste the good stuff into a fresh Issue.

I'll let the maintainers look at your idea now. 😉

**Reply by LeoniePhiline** [reply]
Thanks

**Comment by rarkins** [comment]
I would prefer to keep these grouped per-repo as they not only use separate versions, but one of them is 0.x and the other is 1.x.

**Reply by LeoniePhiline** [reply]
Does that mean you would accept a PR for that?

**Reply by rarkins** [reply]
yes, PRs welcome

**Reply by HonkingGoose** [reply]
We got the PRs:

- #28393
- #28390
